src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensHelpers.cs

```diff
@@ -0,0 +1,243 @@
+                [ClassificationTypeNames.XmlLiteralDelimiter] = LSP.SemanticTokenTypes.Comment, // TO-DO: Potentially change to custom type
+                [ClassificationTypeNames.XmlLiteralEmbeddedExpression] = LSP.SemanticTokenTypes.Comment, // TO-DO: Potentially change to custom type
+                [ClassificationTypeNames.XmlLiteralEntityReference] = LSP.SemanticTokenTypes.Comment, // TO-DO: Potentially change to custom type
+                [ClassificationTypeNames.XmlLiteralName] = LSP.SemanticTokenTypes.Comment, // TO-DO: Potentially change to custom type
+                [ClassificationTypeNames.XmlLiteralProcessingInstruction] = LSP.SemanticTokenTypes.Comment, // TO-DO: Potentially change to custom type
+                [ClassificationTypeNames.XmlLiteralText] = LSP.SemanticTokenTypes.Comment, // TO-DO: Potentially change to custom type
+            };
+
+        /// <summary>
+        /// Returns the semantic tokens for a given document with an optional range.
+        /// </summary>
+        internal static async Task<LSP.SemanticTokens> ComputeSemanticTokensAsync(
+            LSP.TextDocumentIdentifier textDocument,
+            string resultId,
+            string? clientName,
+            ILspSolutionProvider solutionProvider,
+            Dictionary<string, int> tokenTypesToIndex,
+            LSP.Range? range,
+            CancellationToken cancellationToken)
+        {
+            var document = solutionProvider.GetDocument(textDocument, clientName);
+            Contract.ThrowIfNull(document);
+
+            var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+
+            // By default we calculate the tokens for the full document span, although the user 
+            // can pass in a range if they wish.
+            var textSpan = root.FullSpan;
+            if (range != null)
+            {
+                textSpan = ProtocolConversions.RangeToTextSpan(range, text);
+            }
+
+            var classifiedSpans = await Classifier.GetClassifiedSpansAsync(document, textSpan, cancellationToken).ConfigureAwait(false);
+            Contract.ThrowIfNull(classifiedSpans);
+
+            // A TextSpan can be associated with multiple ClassifiedSpans (i.e. if  a token has
+            // modifiers). We perform this group by since LSP requires that each token is
+            // reported together with all its modifiers.
+            var groupedSpans = classifiedSpans.GroupBy(s => s.TextSpan);
+
+            // TO-DO: We should implement support for streaming once this LSP bug is fixed:
+            // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1132601
+            var tokens = ComputeTokens(text.Lines, groupedSpans, tokenTypesToIndex);
+
+            return new LSP.SemanticTokens { ResultId = resultId, Data = tokens };
```

Comment:
resultId is just passed in just to make this object.  just remove the resultId and the `new SemanticTokesn` here.  Just return `tokens`.